Kernel/Syscalls/posix_spawn.cpp

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2025, Tomás Simões <tomasprsimoes@tecnico.ulisboa.pt>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <Kernel/Debug.h>
+#include <Kernel/Devices/BaseDevices.h>
+#include <Kernel/Devices/Device.h>
+#include <Kernel/Devices/Generic/NullDevice.h>
+#include <Kernel/Devices/TTY/TTY.h>
+#include <Kernel/FileSystem/Custody.h>
+#include <Kernel/Memory/Region.h>
+#include <Kernel/Tasks/PerformanceManager.h>
+#include <Kernel/Tasks/Process.h>
+#include <Kernel/Tasks/Scheduler.h>
+#include <Kernel/Tasks/ScopedProcessList.h>
+
+namespace Kernel {
+ErrorOr<FlatPtr> Process::sys$posix_spawn(Userspace<Syscall::SC_posix_spawn_params const*> user_params)
+{
+    VERIFY_PROCESS_BIG_LOCK_ACQUIRED(this);
+    TRY(require_promise(Pledge::exec));
+
+    auto params = TRY(copy_typed_from_user(user_params));
+
+    if (params.arguments.length > ARG_MAX || params.environment.length > ARG_MAX)
+        return E2BIG;
+    if (params.arguments.length == 0)
+        return EINVAL;
+
+    if (params.attr_data.ptr() != 0 || params.attr_data_size != 0 || params.serialized_file_actions_data.ptr() != 0 || params.serialized_file_actions_data_size != 0) {
```

Comment:
Very small nitpick, but the pointers should probably be compared against `nullptr`.

Reply:
`UserSpace<T>::ptr()` returns a `FlatPtr`.